ls1intum/Artemis
# [Bugfix] Questions & Answers: user check and null error on PUT

### Checklist

### Motivation and Context



https://github.com/ls1intum/Artemis/issues/2333

### Description


Access the course via the studentQuestion loaded from the db instead of the studentQuestion passed as body to prevent the course from being null.
Perform the check if the user is the author of a question via a method to cover the case that the user is not loaded on initialization of the component.
Only update question text on PUT to preserve references to lectures and exercises.

### Steps for Testing


1. Log in to Artemis as a *student*
2. Navigate to an exercise or lecture with existing questions or answers
3. Buttons to edit and delete should be shown (if it's your question)
4. Edit the question
5. Saving the changes should not throw any errors

### Test Coverage


unchanged
